crates/bevy_pbr/src/light.rs

```diff
@@ -49,7 +51,7 @@ impl LightRaw {
         let (x, y, z) = global_transform.translation.into();
         LightRaw {
             proj: proj.to_cols_array_2d(),
-            pos: [x, y, z, 1.0],
+            pos: [x, y, z, light.attenuation], // pos.w is the attenuation.
```

Comment:
While possibly more efficient, I'm not sure sticking the attenuation in `pos.w` is actually something we want in the user-facing API. It also seems like an easy mistake in shaders to forget `.xyz`.

Similar for sticking the intensity in `color.a`, although that at least has something of a parallel with alpha. It's also less likely to lead to forgetting `.rgb`, since these are much more often handled separately from `.a`, even if it still would lead to problems.

For reference: Filament does put intensity in `color.w`, but keeps attenuation separate.

Reply:
Correction: The attenuation ends up not affecting the user, since it's read into a separate float in shader code. And then with the padding of a Vec3 when writing to the buffer, it really makes sense to do it this way.

The point about intensity remains.

Reply:
Intensity is now stored separately on the ECS side.